codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/http/ResponseBindingGenerator.kt

```diff
@@ -142,7 +149,13 @@ class ResponseBindingGenerator(protocolConfig: ProtocolConfig, private val opera
                     outputT,
                     errorT
                 ) {
-                    deserializeStreamingBody(binding)
+                    // Streaming unions are Event Streams and should be handled separately
+                    val target = model.expectShape(binding.member.target)
+                    if (target.isUnionShape) {
```

Comment:
I think you can do `target is UnionShape`? and then you probably don't need a cast later since you'll get a smart cast.

Reply:
Fixed.